datetime_formatter/models/res_lang.py

```diff
@@ -105,7 +105,7 @@ def datetime_formatter(self, value, lang=None, template=MODE_DATETIME,
         # Convert str to datetime objects
         if isinstance(value, str):
             try:
-                value = fields.Datetime.from_string(value)
+                value = fields.Datetime.to_datetime(value)
```

Comment:
Why is this change needed? Now Odoo handles natively datetime

Reply:
Look [here](https://github.com/OCA/server-tools/pull/1496/files#diff-3ef157e45961029ea12b27a92d575e19R54). The main usage of this addon, if I don't remember bad, is formatting dates in specific formats for qweb templates, which is sometimes a requirement for website/report design.

In Qweb, you cannot import `fields`, and if an incoming value is a string, you cannot also format it to a datetime object in some contexts, at least not in an easy-to-reuse way.

So, I think that to answer your question, you should answer: when Odoo 12 fetches a date/datetime/time field from database, is it always returned as the corresponding python object? Keep in mind that time fields are usually floats AFAIK, but I haven't had very deep hands-on experience on v12 development yet so as to answer this myself.